src/tools/pub/package-layout.md

```diff
@@ -476,4 +473,33 @@ documentation generators, or other bits of automation.
 Unlike the scripts in `bin`, these are *not* for external users of the package.
 If you have any of these, place them in a directory called `tool`.
 
+## Project specific caching for tools
+
+{{site.alert.info}}
+  Do not check the `.dart_tool/` directory into source control.
+  Instead keep `.dart_tool/` in `.gitignore`.
+{{site.alert.end}}
+
+The `.dart_tool/` is created when you've run `dart pub get`.
+The directory used by various tooling for caching files specific to your project and/or local machine.
+Thus, the `.dart_tool/` directory should never be checked into source control, or copied between machines.
+
+It is also generally safe to delete the `.dart_tool/` directory, though some tools might need recompute the cached information before they work again. 
+
+**Example:** The [dart pub get](https://dart.dev/tools/pub/cmd/pub-get) tool will dowload and extract dependencies to a global `$PUB_CACHE` directory, and then write a `.dart_tool/package_config.json` file mapping _package names_ to directories in the global `$PUB_CACHE` directory. The `.dart_tool/package_config.json` file is used by other tools, such as analyzer, compiler, etc. when they need to resolve statments such as `import "package:foo/foo.dart"`.
```

Comment:
(Break long lines like this. It makes it easier to edit in the future.)

Add comma after "etc.".